changes related to codespaces

KalpanaReddyC opened the pull request:
Below changes are done

1. Added .devcontainer/devcontainer.json file.
2. Renamed Client_Id, Client_Secret to PAYPAL_CLIENT_ID, PAYPAL_CLIENT_SECRET.
3. Updated package.json to include "kill-port" package.

sdarshale commented on .devcontainer/devcontainer.json:
```diff
@@ -0,0 +1,50 @@
+  "image": "mcr.microsoft.com/devcontainers/javascript-node:20",
+  "postCreateCommand": "npm install --ignore-scripts",
+  "postAttachCommand": {
+    "Start server": "npx kill-port 8080 && npm start"
```
adding `--yes` flag would auto-install `kill-port`

sdarshale commented on .devcontainer/devcontainer.json:
```diff
@@ -0,0 +1,50 @@
+      }
+    }
+  }
+}
```
missing newline at end of this file

sdarshale reviewed (changes requested):
- missing newline at the end of some files
- devcontainer file can use `--yes` flag to auto-install `kill-port`
- no package json changes required for `kill-port`. this can be fixed with `--yes` flag in devcontainer file
- asismohanty - for client id and secret changes, please check with domain teams. changing existing code might have negative impact on existing set up.
- need to test `Open In Codespaces` link. it should show options for user inputs on client id secret etc.

KalpanaReddyC commented:
1. missing newline at the end of some files
 - Did not find any file, please specify the filename
2. devcontainer file can use --yes flag to auto-install kill-port
  - I have corrected it by adding "CI=true npx kill-port --yes " in devcontainer.json file
3. no package json changes required for kill-port. this can be fixed with --yes flag in devcontainer file.
  - The above comment is applicable for this also.
4. asismohanty - for client id and secret changes, please check with domain teams. changing existing code might have negative impact on existing set up.
  - This change was asked on slack and so we did that. Reverted back those changes and committed the code.
5. need to test Open In Codespaces link. it should show options for user inputs on client id secret etc.
 - The link in Readme.md file seems to be working perfectly and able to enter values for client-Id and client-secret, if already those values are added then it will not prompt next time.

sdarshale commented on .devcontainer/devcontainer.json:
```diff
@@ -0,0 +1,50 @@
+      "description": "Sandbox Webhook ID of the application.",
+      "documentationUrl": "https://developer.paypal.com/api/rest/webhooks/"
+    },
+    "PAYPAL_API_BASE": {
+      "description": "Sandbox PayPal API base url of the application.",
+      "documentationUrl": "https://api.sandbox.paypal.com/"
+    }
```
We don't need `PAYPAL_API_BASE` in secrets.

sdarshale commented on .devcontainer/devcontainer.json:
```diff
@@ -0,0 +1,50 @@
+    },
+    "PAYPAL_API_BASE": {
+      "description": "Sandbox PayPal API base url of the application.",
+      "documentationUrl": "https://api.sandbox.paypal.com/"
```
We should not put this url in documentationUrl field as users might see it as documentation location, and when they will visit this page, it would return 404.
is there any way to set this url as default value in this secret? if not, we should put this in description and remove documentationUrl.

![image](https://github.com/paypal-examples/p24/assets/141275730/58b86b3a-013c-499c-b45e-ddb8ea3effee)

sdarshale commented on README.md:
```diff
@@ -26,3 +26,6 @@ and configuring your .env config file with your Paypal ClientId and ClientSecret
 4. Navigate to `http://localhost:8080/`
 
+### PayPal Codespaces Links
+[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/im-sandbox-kalpanarc/P24-pay)
```
this one is pointing to your fork

sdarshale replied:
```suggestion
[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/paypal-examples/P24-pay)
```

sdarshale reviewed (approved):
LGTM
